### PR TITLE
ADIS1657X

### PR DESCRIPTION
**Pull request by rbolboac**
- iio_adis: add missing documentation
- add delays after command execution
- remove product id checking
- rename standby flag
- add driver and iio driver support for adis1657x family in already existing adis driver
- add project example for adis1657x

### CODE REVIEW

File: drivers/imu/iio_adis.c [hunk 1]
```diff
@@ -926,8 +926,6 @@ int adis_iio_update_channels(void* dev, uint32_t mask)
 
 	adis = iio_adis->adis_dev;
 
-	iio_adis->active_channels = mask;
-	iio_adis->no_of_active_channels = no_os_hweight32(mask);
```
Comment (amiclaus) [inline drivers/imu/iio_adis.c:930]: unrelated changes.

**Review by amiclaus — COMMENTED** [review]
some minor comments.

**Comment by rbolboac** [comment]
- implemented review changess

File: drivers/imu/adis.c [hunk 2]
```diff
@@ -946,9 +1202,41 @@ int adis_read_y_deltvel(struct adis_dev *adis, int32_t *y_deltvel)
...
 
+/**
+ * @brief Read output FIFO sample count.
+ * @param adis     - The adis device.
+ * @param fifo_cnt - The raw read value.
+ * @return 0 in case of success, error code otherwise.
+ */
+int adis_read_fifo_cnt(struct adis_dev *adis, uint32_t *fifo_cnt)
+{
+	struct adis_field field = adis->info->field_map->fifo_cnt;
```
[the remark this answers is not in the document]
Comment (rbolboac) [reply drivers/imu/adis.c]: yes we can. I made the changes to older lines of code in a new commit and updated this commit where needed.

File: projects/eval-adis1657x/src/platform/stm32/parameters.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+struct stm32_gpio_init_param adis1657x_gpio_reset_extra_ip = {
+	.mode = GPIO_MODE_OUTPUT_OD,
+	.speed = GPIO_SPEED_FREQ_VERY_HIGH,
+};
+
+#ifdef IIO_TRIGGER_EXAMPLE
+struct stm32_gpio_irq_init_param adis1657x_gpio_irq_extra_ip = {
+	.port_nb = 0, /* Port A */
+};
+#endif
```
[the remark this answers is not in the document]
Comment (rbolboac) [reply projects/eval-adis1657x/src/platform/stm32/parameters.c]: done

**Review by buha — APPROVED** [review]
